### PR TITLE
Add CRDs to templates for helm2

### PR DESCRIPTION
**Pull request by shakti-das**
The CRDs location in crds.yaml is still being referenced as `files/` but the files have moved to `crds/`.

### CODE REVIEW

**Comment by shakti-das** [comment]
@howardjohn Follow-up to the crds move. Please review.

**Review by sdake — COMMENTED** [review]
This doesn't quite feel right. I would suggest just deleting this `crds.yaml` file. anything in `templates` will be managed by Istio. The reason the files were moved to `crds` directory is to ensure the helm does not manage the CRDs (and they are not removed on any type of upgrade.

This change would cause them to become managed resources. A managed resource would result in the deletion of the CRs if the CRDs were removed. We want to leak all CRDs on an upgrade.

I haven't looked at the checks, although I would suggest they are failing because there are duplicate CRDs being registered (those but this template, and those by crds.yaml. Simply removing this file would fix that.

Cheers
-steve

**Comment by shakti-das** [comment]
@sdake The checks were failing because of not running `make gen`. 

We (Salesforce) do not use istioctl to install Istio, instead it's Helm + Spinnaker. We are exploring IstioOperator and learned about CRs, but I did not get the point about CRDs if applied via helm deleting CRs. Could you please send me some pointers/references around "A managed resource would result in the deletion of the CRs if the CRDs were removed."

**Review by howardjohn — CHANGES_REQUESTED** [review]
this is meant to support helm3 which reads from CRDs/

**Comment by howardjohn** [comment]
are you using helm2?

**Comment by shakti-das** [comment]
yes, our spinnaker pipeline still uses helm2

**Comment by howardjohn** [comment]
Well we don't want to cause a regression for helm3 users. I suppose we can add this back to templates/ conditionally (off by default). That gives us pretty cheap support for helm2 again. Would that work for you? basically just wrap this with an option

**Comment by shakti-das** [comment]
Yes, that works. I will update the PR accordingly.

File: manifests/charts/base/values.yaml [hunk 1]
```diff
@@ -9,4 +9,6 @@ global:
   istioNamespace: istio-system
 
   istiod:
-    enableAnalysis: false
\ No newline at end of file
+    enableAnalysis: false
+
+enableCRDTemplates: false
```
Comment (howardjohn) [inline manifests/charts/base/values.yaml:14]: Can you put a comment about why you would use this (ie helm2)

**Review by howardjohn — APPROVED** [review]

File: manifests/charts/base/values.yaml (hunk 1 above)
Comment (shakti-das) [reply manifests/charts/base/values.yaml:14]: yes, will add

**Pull request merged**